src/NBitcoin/ChainedBlock.cs

```diff
@@ -10,43 +10,45 @@ namespace NBitcoin
     /// </summary>
     public class ChainedBlock
     {
+        /// <summary>Value of 2^256.</summary>
         private static BigInteger Pow256 = BigInteger.ValueOf(2).Pow(256);
 
+        /// <summary>Window length for calculating median time span.</summary>
         private const int MedianTimeSpan = 11;
 
-        // Pointer to the hash of the block, if any. memory is owned by this CBlockIndex.
-        private uint256 hashBlock;
-        public uint256 HashBlock { get { return this.hashBlock; } }
+        /// <summary>The hash of the block, if any.</summary>
```

Comment:
I'm not sure. I can't see where it will be null. That comment was from C++ code. I removed the ",if any" bit for now as I can't see under what condition it's null.

Reply:
let me check ... so if you go with either constructor, you will have HashBlock calculated for sure and since there is private setter, you can't change it to null ...